Tests/NIOPosixTests/IdleStateHandlerTest.swift

```diff
@@ -180,4 +180,42 @@ class IdleStateHandlerTest : XCTestCase {
         XCTAssertTrue(try channel.finish().isClean)
         eventHandler.assertAllEventsReceived()
     }
+    
+    func testScheduledIdleStateHandlerEventsGetCancelled() throws {
+        final class TestHandler: ChannelInboundHandler {
+            typealias InboundIn = ByteBuffer
+            var events = [IdleStateHandler.IdleStateEvent]()
+            
+            func userInboundEventTriggered(context: ChannelHandlerContext, event: Any) {
+                events.append(event as! IdleStateHandler.IdleStateEvent)
+            }
+        }
+        
+        let embeddedChannel = EmbeddedChannel()
+        let idleStateHandler = IdleStateHandler(
+            readTimeout: .seconds(4),
+            allTimeout: .seconds(5),
+            now: {
+                embeddedChannel.embeddedEventLoop._now
+            }
+        )
+        let testHandler = TestHandler()
+        try embeddedChannel.pipeline.syncOperations.addHandler(idleStateHandler)
+        try embeddedChannel.pipeline.syncOperations.addHandler(testHandler)
+
+        try embeddedChannel.pipeline.syncOperations.connect(
+            to: .init(ipAddress: "127.0.0.1", port: 0),
+            promise: nil
+        )
+        
+        embeddedChannel.embeddedEventLoop.advanceTime(by: .seconds(3))
+        try embeddedChannel.writeInbound(ByteBuffer())
+        
+        embeddedChannel.embeddedEventLoop.advanceTime(by: .seconds(3))
+        try embeddedChannel.pipeline.removeHandler(idleStateHandler).wait()
```

Comment:
I couldn't find a `removeHandler` in `syncOperations` so opted in for a future that we wait.